### DISCUSSION TITLE
Errors when opening Pubchem SDF files

### DISCUSSION THREAD

**Discussion by Unaimend** [rdkit/rdkit#6606] [Q&A] [answered]
Hi, when trying to open a SDF file from Pubchem, I get a lot of errors but I still get valid Mol objects. Is that a problem?

The file in question:

https://drive.google.com/file/d/1kQqSqCqJnVju7L4XVkYgm2jOjXGF0WZ1/view?usp=drive_link

My code 
```Python
import gzip
from rdkit import Chem
from rdkit.Chem import inchi


if __name__ == "__main__":
        suppl = Chem.SDMolSupplier("Substance_000000001_000500000.sdf")
        molecules = [mol for mol in suppl if mol is not None]

    # Now you can work with the molecules, for example, print the number of molecules:
    print("Number of molecules in gzipped SDF file:", len(molecules))
    for i in molecules:
        name = i.GetProp("PUBCHEM_SUBSTANCE_ID")  # Replace "PUBCHEM_COMPOUND_NAME" with the desired property name.
        inchi_str = inchi.MolToInchi(i)
        print(name, inchi_str)
```

**Comment by greglandrum** [comment, marked as answer]
@Unaimend The RDKit often generates warnings, but still returns a useful molecule, when it sees something it doesn't like in an input file. Generally errors result in `None` being returned (i.e. you don't get molecules for those at all), this is why you have the `if mol is not None` check here: `molecules = [mol for mol in suppl if mol is not None]`

Pubchem Substance is a very messy data set, so I would not be at all surprised that you get a large number of both errors and warnings.

**Comment by bp-kelley** [comment]
As an aside, the RDKit's initial structure checker used the pubchem substance for validation tests, and it is as messy as @greglandrum described.

**Comment by Unaimend** [comment]
Thank you for your answers, at least I am little more sure that I am not screwing  up somewhere
